framework/web/User.php

```diff
@@ -562,6 +545,48 @@ protected function sendIdentityCookie($identity, $duration)
     }
 
     /**
+     * Determines if an identity cookie has a valid format and contains a valid auth key. 
+     * This method is used when [[enableAutoLogin]] is true.
+     * This method attempts to authenticate a user using the information in the identity cookie.
+     * @return array|null Returns an array of 'identity' and 'duration' if valid, otherwise null.
+     * @see loginByCookie()
+     */
+    protected function validateIdentityCookie()
+    {
+        $value = Yii::$app->getRequest()->getCookies()->getValue($this->identityCookie['name']);
+        if ($value === null) {
+            return;
+        }
+        $data = json_decode($value, true);
+        if (count($data) === 3 && isset($data[0], $data[1], $data[2])) {
+            list ($id, $authKey, $duration) = $data;
+            /* @var $class IdentityInterface */
+            $class = $this->identityClass;
+            $identity = $class::findIdentity($id);
+            if ($identity !== null) {
```

Comment:
No need to check for `null` since `$identity instanceof IdentityInterface` will return `false` if `$identity` is `null`.